src/group/managed_group/mod.rs

```diff
@@ -102,6 +102,7 @@
         managed_group_config: &ManagedGroupConfig,
         group_id: GroupId,
         key_package_hash: &[u8],
+        extensions: Vec<Box<dyn Extension>>,
```

Comment:
Maybe this should get a builder pattern at some point. It's getting wuite a lot of arguments now and some, like the `extensions` here, might not be needed in many cases (at least in all the tests right now).